lib/ghomfp.gi

```diff
@@ -245,6 +245,11 @@ InstallMethod( ImagesRepresentative, "map from (sub)fp group, rewrite",
     IsMultiplicativeElementWithInverse ], 0,
 function( hom, word )
 local aug,si,r,i,j,tt,ct,cft,c,f,g,ind,e,eval;
+  # catch trivial group
+  if HasMappingGeneratorsImages(hom) 
```

Comment:
In the situation of #4737, the `MappingGeneratorsImages` value is indeed known, and we do not run into the problem anymore after the proposed changed.
My question is: Are we sure that `MappingGeneratorsImages` is set whenever the group in question is trivial?
(Another formulation would be: Are we sure that `CosetTableFpHom` is never called in the case of trivial groups, since its code assumes this?)

Reply:
Also, about the first thing `CosetTableFpHom` does is to invoke `MappingGeneratorsImages`, so perhaps we don't need to check `HasMappingGeneratorsImages` here?

Reply:
The method is installed for `IsFromFpGroupGeneralMappingByImages and IsGroupGeneralMappingByImages`. So `MappingGeneratorsImages` is known. And the error came from accessing the first generator in the `MappingGeneratorsImages` list. So I think this is a perfectly plausible workaround for the concrete error reported.

If your question is more generally: Does every documented operation work OK with a trivial finitely presented group, the answer is probably "no" (and could be another boring issue), and I make no claim to have fixed every case of that.